packages/expo-cli/src/commands/build-native/index.ts

```diff
@@ -1,28 +1,46 @@
-import { BuildType, Platform } from '@expo/build-tools';
+import { Platform } from '@expo/build-tools';
+import { getConfig } from '@expo/config';
 import { User, UserManager } from '@expo/xdl';
 import { Command } from 'commander';
 
 import log from '../../log';
-import Builder, { Options } from './Builder';
+import Builder, { BuilderContext } from './Builder';
 import { printBuildTable } from './utils';
 
-async function buildAction(projectDir: string, options: Options) {
-  if (!options.platform || !Object.values(Platform).includes(options.platform)) {
+async function buildAction(
+  projectDir: string,
+  { platform }: { platform: Platform }
+): Promise<void> {
+  if (!platform || !Object.values(Platform).includes(platform)) {
     throw new Error('Pass valid platform: [android|ios]');
   }
-  const user: User = await UserManager.ensureLoggedInAsync();
-  const builder = new Builder(user);
-  const buildArtifactUrl = await builder.buildProject(projectDir, options);
+  const ctx = await createBuilderContextAsync(projectDir);
+  const builder = new Builder(ctx);
+  const buildArtifactUrl = await builder.buildProjectAsync(platform);
   log(`Artifact url: ${buildArtifactUrl}`);
 }
 
-async function statusAction() {
-  const user: User = await UserManager.ensureLoggedInAsync();
-  const builder = new Builder(user);
-  const result = await builder.getLatestBuilds();
+async function statusAction(projectDir: string): Promise<void> {
+  const ctx = await createBuilderContextAsync(projectDir);
+  const builder = new Builder(ctx);
+  const result = await builder.getLatestBuildsAsync();
   printBuildTable(result.builds);
 }
 
+async function createBuilderContextAsync(projectDir: string): Promise<BuilderContext> {
+  const user: User = await UserManager.ensureLoggedInAsync();
+  const { exp } = getConfig(projectDir);
+  const accountName = exp.owner || user.username;
+  const projectName = exp.slug || 'untitled';
```

Comment:
Are there any repercussions from creating an app named "untitled" on the server (e.g. how hard is it for the user to rename it afterwards? What happens if they already have a different app named `untitled`?). Would it make sense to ask them to give a name if we can't find the name?

Reply:
I sent you a more detailed answer on Slack. I made expo-cli throw an error if `slug` is empty.

Reply:
I've fixed the type of `slug` in `ExpoConfig` here: https://github.com/expo/expo-cli/pull/2260.